server/src/internalClusterTest/java/org/elasticsearch/cluster/PrevalidateNodeRemovalIT.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.cluster;
+
+import org.elasticsearch.action.admin.cluster.node.remove.NodesRemovalPrevalidation;
+import org.elasticsearch.action.admin.cluster.node.remove.PrevalidateNodeRemovalAction;
+import org.elasticsearch.action.admin.cluster.node.remove.PrevalidateNodeRemovalRequest;
+import org.elasticsearch.action.admin.cluster.node.remove.PrevalidateNodeRemovalResponse;
+import org.elasticsearch.test.ESIntegTestCase;
+
+import static org.hamcrest.Matchers.equalTo;
+
+@ESIntegTestCase.ClusterScope(scope = ESIntegTestCase.Scope.TEST)
+public class PrevalidateNodeRemovalIT extends ESIntegTestCase {
+
+    public void testNodeRemovalFromGreenClusterIsSafe() throws Exception {
+        createIndex("test");
+        ensureGreen();
+        String nodeName = randomFrom(internalCluster().getNodeNames());
+        PrevalidateNodeRemovalRequest req = new PrevalidateNodeRemovalRequest(nodeName);
+        PrevalidateNodeRemovalResponse resp = client().execute(PrevalidateNodeRemovalAction.INSTANCE, req).get();
+
+        assertThat(resp.getPrevalidation().getOverallResult().isSafe(), equalTo(NodesRemovalPrevalidation.IsSafe.YES));
+        assertThat(resp.getPrevalidation().getPerNodeResult().size(), equalTo(1));
+        assertThat(resp.getPrevalidation().getPerNodeResult().containsKey(nodeName), equalTo(true));
+        assertThat(resp.getPrevalidation().getPerNodeResult().get(nodeName).isSafe(), equalTo(NodesRemovalPrevalidation.IsSafe.YES));
```

Comment:
NIT: is it possible to replace it with a single assertion? This way in case of failure we would be able to see the entire map content without having to reproduce it.

```
assertThat(resp.getPrevalidation().getPerNodeResult(), equalTo(Map.of(nodeName, NodesRemovalPrevalidation.IsSafe.YES)));
```

I was hoping there is a hamcrest matches for it, but it looks like `hasEntry` permits other entries in map as well.

Reply:
Oh, looks like the entry will contain a reason as well. Was it not asserted intentionally?

Reply:
Yes. For now those details are not clear. I just added that IT to make sure everything works together. There will be more tests and detailed assertions, ofc.